src/nanoarrow/common/inline_array.h

```diff
@@ -972,6 +1101,26 @@ static inline struct ArrowBufferView ArrowArrayViewGetBytesUnsafe(
       view.data.as_uint8 =
           array_view->buffer_views[1].data.as_uint8 + (i * view.size_bytes);
       break;
+    case NANOARROW_TYPE_STRING_VIEW:
+    case NANOARROW_TYPE_BINARY_VIEW: {
+      const union ArrowBufferViewData value_view = array_view->buffer_views[1].data;
+      union ArrowBinaryViewType bvt;
+      const size_t idx = sizeof(union ArrowBinaryViewType) * i;
+      memcpy(&bvt, value_view.as_uint8 + idx, sizeof(union ArrowBinaryViewType));
+      const int32_t inline_size = bvt.inlined.size;
+      view.size_bytes = inline_size;
+      if (inline_size <= NANOARROW_BINARY_VIEW_INLINE_SIZE) {
+        view.data.as_uint8 = value_view.as_uint8 + idx +
+                             sizeof(((union ArrowBinaryViewType*)0)->inlined.size);
+      } else {
+        const int32_t buf_index = bvt.ref.buffer_index;
+        const int32_t nfixed_buf = NANOARROW_BINARY_VIEW_FIXED_BUFFERS;
+        const struct ArrowBufferView variadic_vw =
+            ArrowArrayViewBufferView(array_view, buf_index + nfixed_buf);
+        view.data.as_uint8 = variadic_vw.data.as_uint8 + bvt.ref.offset;
+      }
```

Comment:
I think it'd probably be best to add `const struct ArrowBinaryView* as_binary_view;` to `union ArrowBufferViewData`, and extract a function

```c
static struct ArrowBufferView ArrowArrayViewGetBytesFromViewArrayUnsafe(
    const struct ArrowArrayView* array_view, int64_t i) {
  const struct ArrowBinaryView* bv = &array_view->buffer_views[1].data.as_binary_view[i];
  struct ArrowBufferView out{.size_bytes = bv->inlined.size};
  if (bv->inlined.size <= NANOARROW_BINARY_VIEW_INLINE_SIZE) {
    out.data.as_uint8 = bv->inlined.data;
    return out;
  }

  const int32_t buf_index = bv->ref.buf_index + NANOARROW_BINARY_VIEW_FIXED_BUFFERS;
  out.data.data = array_view->array->buffers[buf_index];
  out.data.as_uint8 += bv->ref.offset;
  return out;
}
```
(replacing ArrowArrayViewBufferView).

 Then we can rewrite this as:
```suggestion
      view = ArrowArrayViewGetBytesFromViewArrayUnsafe(string_array, i);
```

and ArrowArrayViewGetStringUnsafe's case as

```c++
    case NANOARROW_TYPE_STRING_VIEW:
    case NANOARROW_TYPE_BINARY_VIEW: {
      struct ArrowBufferView buf_view = ArrowArrayViewGetBytesFromViewArrayUnsafe(string_array, i);
      view.data = buf_view.data.as_char;
      view.size_bytes = buf_view.size_bytes;
      break;
    }
```

Reply:
Awesome idea - thanks!